src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs

```diff
@@ -43,7 +43,7 @@ public static partial class SymbolFinder
                         if (member != null &&
                             member.IsOverride &&
                             member.OverriddenMember() != null &&
-                           SymbolEquivalenceComparer.Instance.Equals(member.OverriddenMember().OriginalDefinition, symbol.OriginalDefinition))
+                            DependentTypeFinder.OriginalSymbolsMatch(member.OverriddenMember().OriginalDefinition, symbol.OriginalDefinition, solution, cancellationToken))
```

Comment:
The type comparison is already correct or else we wouldn't get to comparing the members yet.